core/src/main/java/com/orientechnologies/orient/core/storage/impl/local/OAbstractPaginatedStorage.java

```diff
@@ -1516,7 +1560,25 @@ public int getClusterIdByName(final String clusterName) {
 
             commitIndexes(indexesToCommit);
 
-            endStorageTx();
+            final OLogSequenceNumber lsn = endStorageTx();
+            final DataOutputStream journaledStream = OAbstractPaginatedStorage.journaledStream;
+            if (journaledStream != null) { // send event to journaled tx stream if the streaming is on
+              final int txId = clientTx.getClientTransactionId();
+              if (lsn == null) // if tx is not journaled
+                try {
+                  journaledStream.writeInt(txId);
+                } catch (IOException e) {
+                  OLogManager.instance().error(this, "unable to write tx id into journaled stream", e);
+                }
+              else
+                writeAheadLog.addEventAt(lsn, () -> {
```

Comment:
Seems like `writeAheadLog` may be `null` at this point if `USE_WAL` set to `false`.

Reply:
fixed